### PR TITLE
[Merged by Bors] - bug fixes and code review comments for the new CI

### PR DESCRIPTION
**Pull request by AmitShaul**
## Motivation
The new CI did not go through a proper integration procedure.
This PR is for a proper code review and new bugs that pop up on the go.

## Result
Removed sensitive information from the code by relaying on new environmental variables.
Fixed a bug when querying for ES IP and getting empty result.
Turned ES and Kibana xpack security on.

### CODE REVIEW

**Comment by AmitShaul** [comment]
bors try

**Comment by AmitShaul** [comment]
bors try

**Comment by AmitShaul** [comment]
bors try-

**Comment by AmitShaul** [comment]
bors try

**Comment by AmitShaul** [comment]
bors try-

**Comment by AmitShaul** [comment]
bors try-

**Comment by AmitShaul** [comment]
bors try-

**Comment by AmitShaul** [comment]
bors try

**Comment by AmitShaul** [comment]
bors try

File: tests/node_pool_deployer.py [hunk 1]
```diff
@@ -26,8 +27,9 @@ def __init__(self, testconfig):
         self.pool_name = f"pool-{self.namespace}"
         self.cluster_name = "spacemesh-cluster-elk"
         self.zone = "us-west1-a"
+        self.project_name = "spacemesh-198810"
```
Comment (lrettig) [inline tests/node_pool_deployer.py]: I still think this (cluster_name, zone, project_name) should not be hardcoded, can we move it into the env?

**Review by lrettig — CHANGES_REQUESTED** [review]

File: tests/node_pool_deployer.py (hunk 1 above)
Comment (AmitShaul) [reply tests/node_pool_deployer.py]: are cluster_name and zone ok or should I remove those as well?

Comment (AmitShaul) [reply tests/node_pool_deployer.py]: fixed

**Comment by AmitShaul** [comment]
bors try

File: tests/config.py [hunk 2]
```diff
@@ -24,9 +12,9 @@ def get_es_password():
 KIBANA_CONF_DIR = './elk/kibana/'
 LOGSTASH_CONF_DIR = './elk/logstash/'
 
-ES_USER_LOCAL = get_es_usr()
-ES_PASS_LOCAL = get_es_password()
-MAIN_ES_IP = "35.197.82.152"
+ES_USER_LOCAL = ut.get_env("ES_USER")
+ES_PASS_LOCAL = ut.get_env("ES_PASS")
+MAIN_ES_IP = "35.227.190.1"
```
Comment (lrettig) [inline tests/config.py]: can we remove the hardcoded IP address here too?

File: tests/context.py [hunk 3]
```diff
@@ -47,6 +47,9 @@ def get_elastic_ip(self, retry=3, interval=1):
 
         for serv in services.items:
             if serv.metadata.name == 'elasticsearch-master':
+                print("#@!#@! serv\n")
```
Comment (lrettig) [inline tests/context.py]: what does this do? maybe add a comment to explain!

File: tests/utils.py [hunk 4]
```diff
@@ -389,3 +389,10 @@ def wrapper(*args, **kwargs):
         end = time.time()
         return result, end-start
     return wrapper
+
+
+def get_env(name, is_must=True):
+    if name not in os.environ and is_must:
+        raise Exception(f"a value must be set for {name} environment variable")
```
Comment (lrettig) [inline tests/utils.py]:
```suggestion
        raise Exception(f"{name} environment variable must be set")
```

**Review by lrettig — CHANGES_REQUESTED** [review]
Looking better!

File: tests/context.py (hunk 3 above)
Comment (AmitShaul) [reply tests/context.py]: this is a debugging print, just so it will be easy for me to spot

File: tests/utils.py (hunk 4 above)
Comment (AmitShaul) [reply tests/utils.py]: fixed

**Comment by AmitShaul** [comment]
bors try

File: tests/context.py (hunk 3 above)
Comment (lrettig) [reply tests/context.py]: I don't think debug output belongs in production... is the plan to remove this before merging?

File: tests/config.py (hunk 2 above)
Comment (AmitShaul) [reply tests/config.py]: fixed

File: tests/context.py (hunk 3 above)
Comment (AmitShaul) [reply tests/context.py]: removed

**Comment by AmitShaul** [comment]
bors try

**Comment by AmitShaul** [comment]
bors try

File: tests/config.py [hunk 5]
```diff
@@ -24,9 +12,9 @@ def get_es_password():
 KIBANA_CONF_DIR = './elk/kibana/'
 LOGSTASH_CONF_DIR = './elk/logstash/'
 
-ES_USER_LOCAL = get_es_usr()
-ES_PASS_LOCAL = get_es_password()
-MAIN_ES_IP = "35.197.82.152"
+ES_USER_LOCAL = ut.get_env("ES_USER")
+ES_PASS_LOCAL = ut.get_env("ES_PASS")
+MAIN_ES_IP = ut.get_env("MAIN_ES_IP")
 MAIN_ES_URL = f"{MAIN_ES_IP}:9200"
```
Comment (lrettig) [inline tests/config.py:18]: Rather than `MAIN_ES_IP` why not pass in the whole `MAIN_ES_URL` including the port too?

File: tests/context.py [hunk 6]
```diff
@@ -46,7 +46,12 @@ def get_elastic_ip(self, retry=3, interval=1):
             retry -= 1
 
         for serv in services.items:
-            if serv.metadata.name == 'elasticsearch-master':
+            # Amit: following a bug - sometimes elasticsearch-master service will be returned before being assigned with
+            # an IP address, in that case run this function again recursively.
+            if serv.metadata.name == 'elasticsearch-master' and not serv.status.load_balancer.ingress:
+                time.sleep(interval)
+                return self.get_elastic_ip()
```
Comment (lrettig) [inline tests/context.py:53]: Can we prevent an infinite loop here? E.g., decrement the `retry` counter each time you make a recursive call, something like that?

**Review by lrettig — APPROVED** [review]
Overall looks fine, see comments I added for small things that could be improved.

**Comment by AmitShaul** [comment]
bors merge

**Comment by AmitShaul** [comment]
bors merge

**Comment by AmitShaul** [comment]
bors try

**Comment by AmitShaul** [comment]
bors try-

**Comment by AmitShaul** [comment]
bors merge

**Comment by AmitShaul** [comment]
bors merge

**Pull request closed without merging**